### PR TITLE
Check components

### PR DESCRIPTION
**Pull request by TammyChen21**
add styling and router for pages, add check-button and counter.

### CODE REVIEW

File: backend/src/main/java/de/neuefische/backend/controller/PlanCardController.java
Comment (lostspezi) [inline backend/src/main/java/de/neuefische/backend/controller/PlanCardController.java:1]: This is an unused class with no content. you can delete it

File: frontend/src/App.tsx [hunk 1]
```diff
@@ -1,58 +1,75 @@
...
+      <Layout>
+         <Routes>
+              <Route path="/" element={<h1>Home</h1>} />
+              <Route path="/plancards" element={<PlanCards plans={plans} addPlan={addPlan}/>}/>
+              <Route path="/plan" element={<PlanList plans={plans} deletePlan={deletePlan} editPlan={editPlan}/>}/>
+              <Route path="/plan/:id" element={<PlanDetailsPage/>}/>
+              <Route path="/profile" element={<Profile/>}/>
+          </Routes>
+      </Layout>
+      </>
```
Comment (lostspezi) [inline frontend/src/App.tsx]:
```suggestion
      <Layout>
         <Routes>
              <Route path="/" element={<h1>Home</h1>} />
              <Route path="/plancards" element={<PlanCards plans={plans} addPlan={addPlan}/>}/>
              <Route path="/plan" element={<PlanList plans={plans} deletePlan={deletePlan} editPlan={editPlan}/>}/>
              <Route path="/plan/:id" element={<PlanDetailsPage/>}/>
              <Route path="/profile" element={<Profile/>}/>
          </Routes>
      </Layout>
```

File: frontend/src/components/PlanItem/PlanItem.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        <div className="plan-item" >
+                    {isEditing ? (
+                        <div>
+                            <input type="text" value={editedDescription}
+                                   onChange={(e) => setEditedDescription(e.target.value)}/>
+                            <button onClick={handleSaveClick} className="save-btn">Save</button>
+                        </div>
+                    ) : (
+                        <div className="edit">
+                            <div className="text" onClick={()=>{ToDetailsPage(plan.id)}} onKeyDown={(e) => { if (e.key === 'Enter') { ToDetailsPage(plan.id) } }}>{plan.description}</div>
```
Comment (lostspezi) [inline frontend/src/components/PlanItem/PlanItem.tsx]: you should not do if statements inside of an element. you should declare a function for this

File: frontend/src/components/Profile/Profile.css
Comment (lostspezi) [inline frontend/src/components/Profile/Profile.css:1]: Why commiting an empty file? delete it

**Review by lostspezi — CHANGES_REQUESTED** [review]
so far so good, few little things, take care of it!;-)

**Review by MarcellNF — CHANGES_REQUESTED** [review]
See my comments above :-)

**Comment by TammyChen21** [comment]
> See my comments above :-)

Thank you! I have resolved all the issues and created a new pull request :-)

**Review by MarcellNF — APPROVED** [review]
LGTM!:)

**Pull request merged**